### PR TITLE
docs: Add why is continuous profiling

### PR DESCRIPTION
**Pull request by Rperry2174**

### CODE REVIEW

File: docs/sources/why-use-continuous-profiling/_index.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+![Diagram showing 3 benefits of continuous profiling](#)
+
+Why prioritize continuous profiling? Here are the key reasons:
+1. **In-Depth Code Insights:** It provides granular, line-level insights into how application code utilizes resources, offering the most detailed view of application performance.
+2. **Complements Other Observability Tools:** Continuous profiling fills critical gaps left by metrics, logs, and tracing, creating a more comprehensive observability strategy.
+3. **Proactive Performance Optimization:** Regular profiling enables teams to proactively identify and resolve performance bottlenecks, leading to more efficient and reliable applications.
+
+## Business Impact of Continuous Profiling
+
+![Infographic illustrating key business benefits](#)
```
Comment (Rperry2174) [inline docs/sources/why-use-continuous-profiling/_index.md:42]: image

File: docs/sources/why-use-continuous-profiling/_index.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+Adopting continuous profiling with tools like Pyroscope can lead to significant business advantages:
+1. **Reduced Operational Costs:** Optimization of resource usage can significantly cut down cloud and infrastructure expenses
+2. **Latency reduction:** Identifying and addressing performance bottlenecks leads to faster and more efficient applications
+3. **Enhanced Incident Management:** Faster problem identification and resolution, reducing Mean Time to Resolution (MTTR) and improving end-user experience
+
+### How Pyroscope helps cut costs
+Pyroscope's low-overhead profiling enables precise optimization of resource usage, directly impacting various cost centers in technology infrastructure. By providing in-depth insights into application performance, Pyroscope allows teams to identify and eliminate inefficiencies, leading to significant savings in areas like observability, incident management, messaging/queuing, deployment tools, and infrastructure.
```
Comment (Rperry2174) [inline docs/sources/why-use-continuous-profiling/_index.md:50]: image

File: docs/sources/why-use-continuous-profiling/_index.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+### How Pyroscope helps reduce latency
```
Comment (Rperry2174) [inline docs/sources/why-use-continuous-profiling/_index.md:52]: image

File: docs/sources/why-use-continuous-profiling/_index.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+Pyroscope plays a pivotal role in reducing application latency by identifying performance bottlenecks at the code level. This granular insight allows for targeted optimization, leading to faster application response times, improved user experience, and consequently, better business outcomes like increased customer satisfaction and revenue.
+
+### How Pyroscope helps with incident management
```
Comment (Rperry2174) [inline docs/sources/why-use-continuous-profiling/_index.md:55]: graphic tbd... I usually like to show an example of when an incident is shown in metrics (i.e. cpu spike) or logs (i.e. OOM errors due to memory leak) 

and then I show how profiling tells you **why** there was a spike in cpu or **where** the memory leak is

File: docs/sources/why-use-continuous-profiling/_index.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+Pyroscope streamlines incident management by offering immediate, actionable insights into application performance issues. With continuous profiling, teams can quickly pinpoint the root cause of an incident, reducing the mean time to resolution (MTTR) and enhancing overall system reliability and user satisfaction.
+
+## Flamegraphs: Visualizing Performance Data
+
+![Example of a flamegraph](#)
```
Comment (Rperry2174) [inline docs/sources/why-use-continuous-profiling/_index.md:60]: image

will need to clean this up and use new grafana color scheme

File: docs/sources/why-use-continuous-profiling/_index.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+Vertically, the nodes in the flamegraph represent the heirarchy of which functions were called and how much time was spent in each function. The top node is the root node and represents the total amount of time spent in the application. The nodes below it represent the functions that were called and how much time was spent in each function. The nodes below those represent the functions that were called from those functions and how much time was spent in each function. This continues until you reach the bottom of the flamegraph.
+
+This is a cpu profile, but profiles can represent many other types of resource such as memory, network, disk, etc. To understand more about how to read a flamegraph, what the different colors mean, and what other types of profiles exist and when to use them see our [flamegraph 101 documention] or [when to use which profile type documentation].
+
+## Pyroscope: Continuous Profiling in Action
+
+![Screenshots of Pyroscope's UI](#)
```
Comment (Rperry2174) [inline docs/sources/why-use-continuous-profiling/_index.md:74]: Show a good diff screenshot

File: docs/sources/profiling-types-in-pyroscope/_index.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
+---
+title: "Understanding Profiling Types and Their Uses in Pyroscope"
+menuTitle: "Understanding Profiling Types in Pyroscope"
+description: "Learn about the different profiling types available in Pyroscope and how to effectively use them in your application performance analysis."
```
Comment (knylander-grafana) [inline docs/sources/profiling-types-in-pyroscope/_index.md:4]: Small stylistic things -- we use [sentence case](https://grafana.com/docs/writers-toolkit/write/style-guide/capitalization-punctuation/) instead of title case. It's also not necessary to use quotes in the front matter of the markdown files. For more information, see https://grafana.com/docs/writers-toolkit/write/front-matter/

```suggestion
title: Understanding profiling types and their uses in Pyroscope
menuTitle: Understanding profiling types
description: Learn about the different profiling types available in Pyroscope and how to effectively use them in your application performance analysis.
```

File: docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
+---
+title: "How to use the Pyroscope UI for performance analysis"
+menuTitle: "How to use the Pyroscope UI"
+description: "How to use the Pyroscope UI to analyze performance of your applications."
```
Comment (knylander-grafana) [inline docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md:4]: Instead of "How to" what about "Use Pyroscope for performance analysis"?  What does UI refer to? The Grafana > Pyroscope plugin UI or something else? Do we need to state UI specifically?

Also, I actually like the folder title. You could have a page that focuses on actually analyzing profiles and a second page that explains the UI. You'd have a task-oriented page and a reference page then. 

Headings should also avoid gerunds, so instead of "Analyzing profiles...", you'd use "Analyze profiles..."

```suggestion
title: Analyze app performance using Pyroscope  
menuTitle: Analyze app performance
description: How to use the Pyroscope UI to analyze performance of your applications.
```

File: docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+weight: 30
+keywords:
+  - pyroscope
+  - UI
+  - performance analysis
+  - flamegraphs
+---
+
+# How to use the Pyroscope UI for performance analysis
```
Comment (knylander-grafana) [inline docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md:13]:
```suggestion
# Analyze app performance using Pyroscope
```

File: docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+## Introduction
```
Comment (knylander-grafana) [inline docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md:15]: Consider removing this heading and instead having a sentence or two introducing the benefit a read will gain from reading through this page and using this content. 

Also, what is the logical progression that someone needs to do to analyze app performance? (If that truly is the purpose  of this page. 

```suggestion
```

File: docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+
+**Understanding Continuous Profiling and Metadata in Pyroscope**
```
Comment (knylander-grafana) [inline docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md:17]: This is actually a heading. Is continuous profiling and metadata unique to Pyroscope? Or is this going from a general discussion of continuous profiling and metadata to a narrower usage with Pyroscope? 
```suggestion
##  Continuous profiling and metadata 
```

File: docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+
+[ TODO ] Add graphic (gif?) showing profiles enriched with of continuous profiling with time and metadata integration
+
+## Key Features of the Pyroscope UI
+
+### Tag Explorer Page
+
+The Tag Explorer page is a vital part of Pyroscope's UI, allowing users to navigate and analyze performance data through tags/labels. This feature is crucial for identifying performance anomalies and understanding the behavior of different application segments under various conditions. We intentionally don't include a query language on this page as we built this page to be as intuitive as possible for users to use the UI to navigate and drill down into which tags are most interesting to them.
+
+**How to Use the Tag Explorer:**
```
Comment (knylander-grafana) [inline docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md:36]:
```suggestion
To use the Tag Explorer:
```

File: docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+**Understanding Continuous Profiling and Metadata in Pyroscope**
+
+While code profiling has been a long-standing practice, Continuous Profiling represents a modern and more advanced approach to performance monitoring. This technique adds two critical dimensions to traditional profiles:
+
+- **Time:** Profiling data is collected _continuously_, providing a time-centric view that allows querying performance data from any point in the past
+- **Metadata:** Profiles are enriched with metadata, adding contextual depth to the performance data
+
+These dimensions, coupled with the detailed nature of performance profiles, make Continuous Profiling a uniquely valuable tool. Pyroscope's UI enhances this further by offering a convenient platform to analyze profiles and get insights that are impossible to get from using other traditional signals like logs, metrics, or tracing. 
+
+In this walkthrough, we'll show how Pyroscope parallels these other modern observability tools by providing a Prometheus-like querying experience. More importantly, you'll learn how to use Pyroscope's extensive UI features for a deeper insight into your application's performance.
```
Comment (knylander-grafana) [inline docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md:26]: This entire section has great information, but it says that "in this walkthrough" -- except what follows is a UI reference page and a Pyroscope in Grafana page. Does this content really belong as an introduction ot the Why use continuous profiling? page? 

What would you think about breaking this page into multiple sections? It feels like this page is trying to do several things. It might be easier for a user to understand as separate pages. 

1. Pyroscope UI overview (see examples, [Grafana dashboards overview](https://grafana.com/docs/grafana/latest/fundamentals/dashboards-overview/) and [Panel editor overview](https://grafana.com/docs/grafana/latest/panels-visualizations/panel-editor-overview/) 
2.  Pyroscope in Grafana - as separate page that provides an example of what Pyroscope in Grafana provides  (see example, [Tempo in Grafana](https://grafana.com/docs/tempo/latest/getting-started/tempo-in-grafana/)  
3. Continuous profiling content - either move to it's own page or add to why use continuous profilng page as an intro

File: docs/sources/why-use-continuous-profiling/_index.md (hunk 1 above)
Comment (knylander-grafana) [reply docs/sources/why-use-continuous-profiling/_index.md:42]: Instructions for adding images: https://grafana.com/docs/writers-toolkit/write/image-guidelines/

**Comment by knylander-grafana** [comment]
Hi, Ryan. 

Great start on the content.The PR seems to address multiple issues in your PR. I added the ones I think you're address to the description. Those issues have the outlines and notes from my doc review earlier. They may help give some additional ideas. 

Unfortunately, I'm out of time today so I didn't have a chance to complete my comments on your doc. This is an early draft so there's a lot more you probably want to do. I'm happy to continue reviewing when I get back. 

Take a look at the information architecture in this image. It will give you an idea of the proposed information architecture for the Pyroscope doc. 
image

The content that you have in this PR aligns with the Introduction content and the "use Pyroscope" content - Analzye and Visualize profile data (see #2776). We may want to move some of the pages around so they match the information architecture. 

Thank you for contributing to doc!

File: docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md (hunk 13 above)
Comment (Rperry2174) [reply docs/sources/analyzing-profiles-in-pyroscope-ui/_index.md:25]: I see yeah I'll rephrase it to say UI reference. I will create an issue for this as I think I'd need to see this to see what it would look like split up

**Review by petethepig — APPROVED** [review]
